src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStream.cs

```diff
@@ -61,6 +63,8 @@ internal class TdsStream : Stream, ITdsWriteStream, ITdsReadStream
         /// <inheritdoc />
         public byte ReadPacketHeaderType => _readStream.ReadPacketHeaderType;
 
+        public TdsWriter GetTdsWriter() => _writer;
```

Comment:
This can be an autoproperty with private setter

```suggestion
        public TdsWriter TdsWriter { get; private set; }
```